src/REP15Utils.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+library REP15Utils {
+  struct Delegation {
+    address delegatee;
+    uint64 until;
+    bool delegated;
+  }
+
+  struct Context {
+    address controller;
+    uint64 detachingDuration;
+  }
+
+  struct TokenContext {
+    bool attached;
+    bool locked;
+    address user;
+    uint64 readyForDetachmentAt;
+  }
```

Comment:
can utilize pattern like:
```solidity
using REP15Utils for Delegation global;
using REP15Utils for Context global;
using REP15Utils for TokenContext;
``` 
You can reduce redundant code using for in utilizing contract.

Reply:
thank you, but why TokenContext not global?

Reply:
updated.